src/GHCMod/Options/Commands.hs

```diff
@@ -0,0 +1,230 @@
+module GHCMod.Options.Commands where
+
+import Options.Applicative
+import Options.Applicative.Types
+import Language.Haskell.GhcMod.Types
+import GHCMod.Options.DocUtils
+import Language.Haskell.GhcMod.Read
+
+type Symbol = String
+type Expr = String
+type Module = String
+type Line = Int
+type Col = Int
+type Point = (Line, Col)
+
+data GhcModCommands =
+    CmdLang
+  | CmdFlag
+  | CmdDebug
+  | CmdBoot
+  | CmdNukeCaches
+  | CmdRoot
+  | CmdLegacyInteractive
+  | CmdModules Bool
+  | CmdDumpSym FilePath
+  | CmdFind Symbol
+  | CmdDoc Module
+  | CmdLint LintOpts FilePath
+  | CmdBrowse BrowseOpts [Module]
+  | CmdDebugComponent [String]
+  | CmdCheck [FilePath]
+  | CmdExpand [FilePath]
+  | CmdInfo FilePath Symbol
+  | CmdType FilePath Point
+  | CmdSplit FilePath Point
+  | CmdSig FilePath Point
+  | CmdAuto FilePath Point
+  | CmdRefine FilePath Point Expr
+
+commandsSpec :: Parser GhcModCommands
+commandsSpec =
```

Comment:
I'm a bit nitpicky about too many brackets in the option parsers I've come up with some cusom infix versions of the operators needed to make it all work out without brackets: https://github.com/DanielG/kvm-in-a-box/blob/master/src/Options.hs#L67